src/scanner.c

```diff
@@ -3091,8 +3110,7 @@ static Int InitKernel (
     Int                 i;
 
     TLS(Input) = TLS(InputFiles);
-    TLS(Input)--;
-    (void)OpenInput(  "*stdin*"  );
+    (void)OpenInput_Internal(  "*stdin*", 1  );
```

Comment:
I am very slightly concerned about the second part of this PR. It seems to replace one fragile with another fragile solution. In particular, this here is the only place where `FirstOpen` is ever set to true / 1, right? So, anybody who refactors the code must be aware that this always must remain the first place to open an input stream; and we must make sure that *stdin* is never ever closed (and then perhaps reopened later)... correct? Or perhaps this is impossible anyway? Still, though: The reasons why this code change is "safe" depend on knowledge about the whole of the GAP kernel, which is why I call them "fragile". I'd much prefer a solution that is "intrinsically" correct... 

On the long run, I think the solution would be to simply rewrite the whole "input stack" code. On the short term, why don't we simpy does what `OpenOutput` does to solve this? I.e. initialize `TLS(Input)` to 0, then teach `OpenInput` and `OpenInputStream` to work with that...